### PR TITLE
Add wording about output_data in case of output_size=0

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -419,6 +419,8 @@ struct evmc_result
      *  freed with evmc_result::release().
      *
      *  This MAY be NULL.
```
Comment (axic) [inline include/evmc/evmc.h:421]: Why is this, and some other places double indented, but not every block?

**Comment by axic** [comment]
Oh no, the doxygen link in CI is 404, hence the failing test.

File: include/evmc/evmc.h (hunk 1 above)
Comment (chfast) [reply include/evmc/evmc.h:421]: Because of migration from `///` comments to `/**` long time ago.

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -419,6 +419,8 @@ struct evmc_result
...
+     *
+     * If output_size is NULL this pointer MAY NOT be dereferenced.
```
Comment (chfast) [inline include/evmc/evmc.h:423]:
```suggestion
     * If result::output_size is 0 this pointer MUST NOT be dereferenced.
```
and combine with the paragraph above.

File: include/evmc/evmc.h (hunk 1 above)
Comment (axic) [reply include/evmc/evmc.h]: Fixed.

**Review by chfast — APPROVED** [review]

**Comment by axic** [comment]
New build here: https://output.circle-artifacts.com/output/job/49b296b4-94bd-44c8-a556-f51800483554/artifacts/0/docs/index.html

Screenshot

**Pull request merged**